django/contrib/admin/helpers.py

```diff
@@ -342,7 +342,9 @@ def __iter__(self):
             )
 
         for form, original in zip(
-            self.formset.initial_forms, self.formset.get_queryset()
+            self.formset.initial_forms,
+            self.formset.get_queryset()[:self.formset.initial_form_count()],
```

Comment:
We only need as many results as the number of initial form forms.
This may be more optimal if the queryset were to return a larger number of results.